model/src/items/skins.rs

```diff
@@ -1 +1,150 @@
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+use crate::items::{
+    ArmorSlot, DamageType, GatheringToolsType, Rarity, Restrictions, WeaponType, WeightClass,
+};
+use crate::misc::colors::ColorId;
+use crate::{BulkEndpoint, Endpoint, EndpointWithId};
+
 pub type SkinId = u32;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Debug, Serialize, Deserialize)]
+#[cfg_attr(test, serde(deny_unknown_fields))]
+pub enum SkinType {
+    Armor,
+    Back,
+    Gathering,
+    Weapon,
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Serialize, Deserialize)]
+#[cfg_attr(test, serde(deny_unknown_fields))]
+pub enum Flags {
+    /// Displayed in the account wardrobe.
+    ShowInWardrobe,
+    /// Applying the skin is free.
+    NoCost,
+    /// The skin is hidden until it is unlocked.
+    HideIfLocked,
+    /// The skin overrides item rarity when applied.
+    OverrideRarity,
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Debug, Serialize, Deserialize)]
+#[cfg_attr(test, serde(deny_unknown_fields))]
+pub enum Material {
+    #[serde(rename = "cloth")]
+    Cloth,
+    #[serde(rename = "leather")]
+    Leather,
+    #[serde(rename = "metal")]
+    Metal,
+}
```

Comment:
You can use `serde(rename_all = "lowercase")` here

Reply:
Not thought of this one, better indeed !